src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -625,7 +625,7 @@ private bool IsPossibleSubpatternElement()
             bool isExpression = this.IsPossibleExpression() &&
                     // IsPossibleExpression returns true when the next token is a binary operator.
                     // That is useful for error recovery elsewhere, but not here.
-                    !(SyntaxFacts.IsBinaryExpression(tk) || SyntaxFacts.IsAssignmentExpressionOperatorToken(tk));
+                    !(SyntaxFacts.IsBinaryExpression(tk) && !SyntaxFacts.IsPrefixUnaryExpression(tk) || SyntaxFacts.IsAssignmentExpressionOperatorToken(tk));
```

Comment:
fwiw, this expression has now reached the point of not being comprehensible to me in this compact form.

Reply:
Same for me

Reply:
Turns out this whole expression can be replaced by a call to a recently introduced helper (actually, the same `IsPossibleExpression` helper with new parameters).